packages/mgt-components/src/components/mgt-file-list/mgt-file-upload/mgt-file-upload.ts

```diff
@@ -1195,14 +1197,17 @@ export class MgtFileUpload extends MgtBaseComponent {
           }
         }
         continue;
-      } else {
-        const fileItem = uploadFileItem.getAsFile();
-        if (fileItem) {
-          this.writeFilePath(fileItem, '');
-          collectFilesItems.push(fileItem);
+      }
+
+      if (fileType) {
+        uploadFileItem = uploadFileItem as File;
```

Comment:
Let's keep the use of `getAsFile` this will enforce that the object is falsy if it's not a File kind. Using a TypeScript cast here is a compile time only check, while the method provides runtime safety

```suggestion
        uploadFileItem = uploadFileItem.getAsFile();
```

Reply:
That was the cause of the bug. A `File` object doesn't have a `getAsFile()` method and errored when you upload using the button. The upload button flow gives a `FileList` iterator which gives us the required `File` therefore no need for "getting" the file. Is there a different runtime fix other than doing it at compile time with typecasting in this scenario?

Reply:
Oh, thanks for pointing out my mistake here, in that case we should be able to use a type guard to ensure that we have a File object